hs_tools_resource/utils.py

```diff
@@ -58,3 +67,58 @@ def get_SupportedSharingStatus_choices():
             ['Discoverable', 'Discoverable'],
             ['Private', 'Private'],
             ]
+
+
+def copy_res_to_specified_federated_irods_server_as_needed(app_shortkey, res_shortkey, user):
+    """
+    When app resource has iRODS federation target path and target resource defined as
+    extended metadata, the resource needs to be pushed to the specified iRODS federation path
+    in the specified iRODS storage resource.
+    :param app_shortkey: shortkey of the app tool resource to be launched
+    :param res_shortkey: shortkey of the resource launching the app tool resource
+    :param user: requesting user
+    :return:
+    """
+    # check whether irods_path_key and irods_resc_key are added as extended metadata of the
+    # app tool resource, and if they are, push resource to specified iRODS target accordingly
+    filterd_app_obj = BaseResource.objects.filter(short_id=app_shortkey).filter(
+        extra_metadata__has_key=irods_path_key).filter(
+        extra_metadata__has_key=irods_resc_key).first()
+    if filterd_app_obj:
+        try:
+            res = get_resource_by_shortkey(res_shortkey)
+            app_res = get_resource_by_shortkey(app_shortkey)
+            user_auth_key_exist = BaseResource.objects.filter(short_id=app_shortkey).filter(
+                extra_metadata__has_key=user_auth_flag_key).first()
+            if user_auth_key_exist:
+                req_user_auth = True if app_res.extra_metadata[user_auth_flag_key].lower() == 'true' \
```

Comment:
I think it's worth pointing out that this will resolve False if there is a value other than true (case insensitive).  I'd prefer just looking to see if this key exists and not evaluate the value.  I don't feel strongly on this, as long as the value necessary is documented.

Reply:
@sblack-usu Good point. However, if we don't evaluate the value, the user could put a ```False``` for the key, and we would treat it as true which does not sound right to me. We will definitely need to document it in user documentation. I have noted that and will emphasize this when contacting @Lizabrazil for additional documentation introduced by this PR. I will keep it as is now .